overlord/devicestate/handlers_install.go

```diff
@@ -313,25 +331,36 @@ func checkEncryption(model *asserts.Model) (res bool, err error) {
 		return false, nil
 	}
 
-	// encryption is required in secured devices and optional in other grades
-	if err := secbootCheckKeySealingSupported(); err != nil {
+	// check if the model prefers to be unencrypted
+	// TODO: provide way to select via install chooser menu
+	//       if the install is unencrypted or encrypted
+	if model.StorageSafety() == asserts.StorageSafetyPreferUnencrypted {
+		logger.Noticef(`installing system unencrypted to comply with prefer-unencrypted storage-safety model option`)
+		return false, nil
+	}
+
+	// check encryption support now
+	var (
+		encryptionViaHookErr    error
+		encryptionViaSecbootErr error
+	)
+	// first check fde-setup hook based encryption
+	encryptionViaHookErr = hooksCheckKeySealingSupported(st, deviceCtx)
+	// if hooks did not work, check build-in seccboot support
+	if encryptionViaHookErr != nil {
```

Comment:
I still think that if the hook is present we shouldn't try to check for the tpm even if it failed